runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -6482,6 +6482,51 @@ TR::Register *J9::Power::TreeEvaluator::VMnewEvaluator(TR::Node *node, TR::CodeG
             genInitArrayHeader(node, iCursor, isVariableLen, clazz, NULL, resReg, zeroReg, condReg, enumReg, dataSizeReg,
                   tmp5Reg, tmp4Reg, conditions, needZeroInit, cg);
 
+#ifdef TR_TARGET_64BIT
+         TR::Register *offsetReg = tmp5Reg;
+         TR::Register *firstDataElementReg = tmp4Reg;
+         TR::MemoryReference *dataAddrSlotMR = NULL;
+
+         if (TR::Compiler->om.compressObjectReferences() && isVariableLen)
+            { // We need to check sizeReg at runtime to determine correct offset of dataAddr field.
```

Comment:
Let's put comments either before the if or inside the block. Once we do get to some automated formatting this will not play nice with tooling.